grammars/julia.cson

```diff
@@ -124,7 +126,7 @@ repository:
             name: "entity.name.function.julia"
           "2":
             name: "support.type.julia"
-        match: "([\\w.!]+)({[^}]*})?\\((?=.*\\)\\s*=(?!=))"
+        match: "([\\w.!]+)({(?:[^{}]|{(?:[^{}]|{[^{}]*})*})*})?\\((?=.*\\)\\s*=(?!=))"
         comment: "TODO: why do we have `.` in the function name characer class?"
```

Comment:
I think we should scrap the `.` in the first group, like we did in the function call repo